### PR TITLE
refactor(site): refactor login screen

### PR DESCRIPTION
**Pull request by BrunoQuaresma**
Before:
Screenshot 2023-11-17 at 15 08 02

After:
Screenshot 2023-11-17 at 15 08 12

### CODE REVIEW

**Comment by matifali** [comment]
I think we should keep the username password fields collapsed when an OAuth login is available.

**Comment by BrunoQuaresma** [comment]
Honestly, I never saw this pattern before and I find it a bit strange. If email/password is not important, it can be disabled. Make the user have to click in a button to show the form that could be displayed right away, does not sound good to me.

**Comment by matifali** [comment]
@BrunoQuaresma, I am okay with the new design, too. It does look more premium ❤️. I have seen the experience I describe with other self-hosted enterprise tools, e.g., Artifactory.

**Review by buenos-nachos — APPROVED** [review]
Looks good! Just had one question for my own understanding, and one question about how the HTML semantics are set up

File: site/src/@types/mui.d.ts [hunk 1]
```diff
@@ -31,6 +31,10 @@ declare module "@mui/material/Button" {
   interface ButtonPropsColorOverrides {
     neutral: true;
   }
+
+  interface ButtonPropsSizeOverrides {
+    xlarge: true;
```
[the remark this answers is not in the document]
Comment (BrunoQuaresma) [reply site/src/@types/mui.d.ts:36]: Exactly!

File: site/src/pages/LoginPage/OAuthSignInForm.tsx [hunk 2]
```diff
@@ -26,49 +25,47 @@ export const OAuthSignInForm: FC<OAuthSignInFormProps> = ({
   return (
     <Box display="grid" gap="16px">
       {authMethods?.github.enabled && (
-        <Link
+        <Button
```
[the remark this answers is not in the document]
Comment (BrunoQuaresma) [reply site/src/pages/LoginPage/OAuthSignInForm.tsx:28]: `Link` works well when doing internal navigation to handle react-router state changes but when it is external, I don't see why to use a Link.

Ps: Link when it is from `import { Link } from "react-router-dom"`.

I use `Link` from MUI when I want to style it as a link with the blue color and decoration on hover.

**Pull request merged**